tasks/main.yml

```diff
@@ -16,12 +16,14 @@
 - name: Install and configure required providers
   vars:
     __certificate_providers: |
-      {{
-        certificate_requests |
-        map(attribute='provider', default=__certificate_provider_default) |
-        unique |
-        list
-      }}
+      {% set ns = namespace(unique_providers=[]) %}
+      {% for item in certificate_requests %}
+      {%   set _ = ns.unique_providers.append(
```

Comment:
Why to we have to use `set _` here?

Reply:
not sure - I think in jinja2 you cannot have a "bare expression" e.g. something like
```python
for item in certificate_requests:
    ns.unique_providers.append(item.provider)
```
would be perfectly legal in python, but apparently jinja2 considers `ns.unique_providers.append(item.provider)` by itself to not be a valid statement, so you have to have the `set` with the "dummy" assignment to make it a real jinja2 statement.  We also ran into this with logging and solved it the same way.  If you can figure out a better way to do this, please do.